http/middleware.go

```diff
@@ -46,6 +46,43 @@ var (
 	colContextKey = contextKey("col")
 )
 
+// mustGetContextClientCollection returns the client collection from the http request context or panics.
+//
+// This should only be called from functions within the http package.
+func mustGetContextClientCollection(req *http.Request) client.Collection {
+	return req.Context().Value(colContextKey).(client.Collection) //nolint:forcetypeassert
+}
```

Comment:
thought: It feels wrong to have these in the `middleware` file. They should probably be moved to the `utils` file instead including the context variables above.

Reply:
I agree with this thought.

(1) I was tempted to perhaps have a shared `utils` files some how for both `http` and `cli` as cli implements some of these methods too (however will have to make `http` and `cli` under a common parent folder perhaps that has the common `utils` file but we already have a `client` folder which I feel a bit uneasy moving both these into, perhaps another folder?). 

(2) Another approach might be to do what you suggested in a `utils` file similar to how it's done in `cli` package and call it a day.

Can follow consensus, does anyone else have any suggestions/prefs? @islamaliev @nasdf @AndrewSisley

Reply:
I prefer (2) :)

Reply:
- [ ] Do (2)

Reply:
Resolved in commit: `PR(ADHERE): Adhere to code review - move to utils`